interfaces/hotplug/deviceinfo.go

```diff
@@ -89,3 +90,43 @@ func (h *HotplugDeviceInfo) Attribute(name string) (string, bool) {
 	val, ok := h.Data[name]
 	return val, ok
 }
+
+func (h *HotplugDeviceInfo) firstAttrValueOf(tryAttrs ...string) string {
+	for _, attr := range tryAttrs {
+		if val, ok := h.Attribute(attr); ok && val != "" {
+			return val
+		}
+	}
+	return ""
+}
+
+func (h *HotplugDeviceInfo) String() string {
+	var str []string
+
+	if devname := h.DeviceName(); devname != "" {
+		str = append(str, fmt.Sprintf("devname:%s", devname))
+	} else {
+		if devpath := h.DevicePath(); devpath != "" {
+			str = append(str, fmt.Sprintf("devpath:%s", devpath))
+		}
+	}
+	for _, attr := range []string{"MAJOR", "MINOR"} {
+		if val, ok := h.Attribute(attr); ok {
+			str = append(str, fmt.Sprintf("%s:%s", strings.ToLower(attr), val))
```

Comment:
Why `ToLower`?

Reply:
Just personal preference/feeling, I liked the lowercased attribute names more, especially as it also goes into task summaries. Again, happy to bikeshed / change ;)

Reply:
That's quite okay, please just add a comment.